src/lib.rs

```diff
@@ -52,37 +51,38 @@ impl<'a> MownStr<'a> {
         };
         MownStr {
             addr,
-            xlen: other.len(),
+            len: other.len(),
             _phd: PhantomData,
+            owners: None,
         }
     }
+    fn cleanup(&self) -> bool {
+        self.owners
+            .as_ref()
+            .map(|c| c.load(Ordering::SeqCst) == 0)
+            .unwrap_or(false)
+    }
 
     pub const fn is_borrowed(&self) -> bool {
-        (self.xlen & OWN_FLAG) == 0
+        self.owners.is_none()
     }
 
     pub const fn is_owned(&self) -> bool {
-        (self.xlen & OWN_FLAG) == OWN_FLAG
+        self.owners.is_some()
     }
 
-    pub const fn borrowed(&self) -> MownStr {
-        MownStr {
-            addr: self.addr,
-            xlen: self.xlen & LEN_MASK,
-            _phd: PhantomData,
-        }
-    }
```

Comment:
@pchampin  this method was particularly finicky in triggering use after free UB behaviour in a lot of [sophia term methods](https://github.com/pchampin/sophia_rs//blob/d5be5334e0f76eebd8263e37870b46ebfb7738f6/api/src/term/_simple.rs#L48)

I strongly recommend removing it altogether and rely on `Clone` + `Arc` semantics to conserve on string allocation